biz.aQute.bndlib/src/aQute/bnd/osgi/resource/CapReq.java

```diff
@@ -48,14 +51,17 @@ public Resource getResource() {
 
 	@Override
 	public int hashCode() {
+		if (hashCode != 0) {
```

Comment:
Yes, in the pathological case that the actual computed has value is zero, then it will recompute every time. So what? The odds are probably 1 in 2^32 of that happening :-) So, in any practical sense, it will not happen even close to enough to be any issue.